pkg/infrastructure/services/messaging_knative.go

```diff
@@ -0,0 +1,162 @@
+// limitations under the License.
+
+package services
+
+import (
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	sourcesv1alpha1 "knative.dev/eventing/pkg/apis/sources/v1alpha1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	alpha1 "knative.dev/pkg/apis/duck/v1alpha1"
+	"knative.dev/pkg/tracker"
+)
+
+const (
+	topicIdentifier = "kogito.kie.org/messageEventId"
+)
+
+// knativeMessagingDeployer implementation of messagingHandler
+type knativeMessagingDeployer struct {
+	messagingDeployer
+}
+
+func (k *knativeMessagingDeployer) createRequiredResources(service v1alpha1.KogitoService) error {
+	infra, err := k.fetchInfraDependency(service, infrastructure.IsKnativeEventingResource)
+	if err != nil || infra == nil {
+		return err
+	}
+
+	// since we depend on Knative, let's bind a SinkBinding object to our deployment
+	sinkBinding := k.newSinkBinding(service, infra)
+	if _, err := kubernetes.ResourceC(k.cli).CreateIfNotExistsForOwner(sinkBinding, service, k.scheme); err != nil {
+		return err
+	}
+
+	// fetch for consumed topics to create our triggers
+	topics, err := k.fetchRequiredTopics(service)
+	if err != nil {
+		return err
+	}
+
+	var knativeRes resource.KubernetesResource
+	exists := false
+	for _, topic := range topics {
+		if topic.Kind == consumed {
+			if exists, err = k.triggerExists(topic, service); err != nil {
+				return err
+			} else if !exists {
+				knativeRes = k.newTrigger(topic, service, infra)
+				if err := kubernetes.ResourceC(k.cli).CreateForOwner(knativeRes, service, k.scheme); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// newTrigger creates a new Knative Eventing Trigger reference for the given Topic
+func (k *knativeMessagingDeployer) newTrigger(t messageTopic, service v1alpha1.KogitoService, infra *v1alpha1.KogitoInfra) *eventingv1.Trigger {
+	return &eventingv1.Trigger{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-listener-%s", service.GetName(), util.RandomSuffix()),
+			Namespace: service.GetNamespace(),
+			Labels: map[string]string{
+				framework.LabelAppKey: service.GetName(),
+				topicIdentifier:       t.Name,
+			},
+		},
+		Spec: eventingv1.TriggerSpec{
+			Broker: infra.Spec.Resource.Name,
+			Filter: &eventingv1.TriggerFilter{Attributes: eventingv1.TriggerFilterAttributes{"type": t.Name}},
+			Subscriber: duckv1.Destination{
+				Ref: &duckv1.KReference{
+					Name:       service.GetName(),
+					Namespace:  service.GetNamespace(),
+					Kind:       "Service",
+					APIVersion: "v1",
+				},
```

Comment:
Use `APIVersion: corev1.SchemeGroupVersion.Version` to drive APIVersion for `Service`.